From me@felipebalbi.com  Tue Dec 16 13:47:10 2008
From: Robin Getz <rgetz@blackfin.uclinux.org>
Date: Tue,  2 Dec 2008 21:33:45 +0200
Subject: USB: musb: Make sure we program the correct values in only when necessary.
Cc: Greg Kroah-Hartman <gregkh@suse.de>, Robin Getz <rgetz@blackfin.uclinux.org>, Bryan Wu <cooloney@kernel.org>, Felipe Balbi <felipe.balbi@nokia.com>
Message-ID: <1228246431-12816-4-git-send-email-me@felipebalbi.com>

From: Robin Getz <rgetz@blackfin.uclinux.org>

Make sure we program the correct values in only when necessary.

Signed-off-by: Robin Getz <rgetz@blackfin.uclinux.org>
Signed-off-by: Bryan Wu <cooloney@kernel.org>
Signed-off-by: Felipe Balbi <felipe.balbi@nokia.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/musb/blackfin.c |   14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

--- a/drivers/usb/musb/blackfin.c
+++ b/drivers/usb/musb/blackfin.c
@@ -245,13 +245,15 @@ int __init musb_platform_init(struct mus
 	}
 	gpio_direction_output(musb->config->gpio_vrsel, 0);
 
-	/* Anomaly #05000346 */
-	bfin_write_USB_APHY_CALIB(0x5411);
-	SSYNC();
+	if (ANOMALY_05000346) {
+		bfin_write_USB_APHY_CALIB(ANOMALY_05000346_value);
+		SSYNC();
+	}
 
-	/* Anomaly #05000347 */
-	bfin_write_USB_APHY_CNTRL(0x0);
-	SSYNC();
+	if (ANOMALY_05000347) {
+		bfin_write_USB_APHY_CNTRL(0x0);
+		SSYNC();
+	}
 
 	/* TODO
 	 * Set SIC-IVG register
